src/assembly/stack.ts

```diff
@@ -100,93 +99,119 @@ export class StackManifest {
 
     /**
      * Get the CloudFormation template fragment of the resource with the given
-     * logicalId
+     * logicalId in the given stack
      *
+     * @param stackPath - The path to the stack
      * @param logicalId - The CFN LogicalId of the resource
      * @returns The resource portion of the CFN template
      */
-    public resourceWithLogicalId(logicalId: string): CloudFormationResource {
-        if (logicalId in this.resources) {
-            return this.resources[logicalId];
+    public resourceWithLogicalId(stackPath: string, logicalId: string): CloudFormationResource {
+        const stackTemplate = this.stacks[stackPath];
+        if (!stackTemplate) {
+            throw new Error(`Could not find stack template for path ${stackPath}`);
+        }
+        const resourcesToSearch = stackTemplate.Resources ?? {};
+        if (logicalId in resourcesToSearch) {
+            return resourcesToSearch[logicalId];
         }
         throw new Error(`Could not find resource with logicalId '${logicalId}'`);
     }
 
     /**
-     * Get the CloudFormation template fragment of the resource with the given
-     * CDK construct path
+     * Get the nested stack path from the path of the nested stack resource (i.e. 'AWS::CloudFormation::Stack').
+     * For Nested Stacks, there's two nodes in the tree that are of interest. The first node is the `AWS::CloudFormation::Stack` resource,
+     * it's located at the path `parent/${NESTED_STACK_NAME}.NestedStack/${NESTED_STACK_NAME}.NestedStackResource`.
+     * This is the input to the function. The second node houses all of the children resources of the nested stack.
+     * It's located at the path `parent/${NESTED_STACK_NAME}`. This is the the return value of the function.
      *
-     * @param path - The construct path to find the CFN Resource for
-     * @returns The resource portion of the CFN template
-     */
-    public resourceWithPath(path: string): CloudFormationResource {
-        const logicalId = this.logicalIdForPath(path);
-        if (logicalId && logicalId in this.resources) {
-            return this.resources[logicalId];
+     * The tree structure looks like this:
+     * ```
+     * Root
+     * ├── MyNestedStack.NestedStack
+     * │   └── AWS::CloudFormation::Stack (Nested Stack Resource)
+     * │       └── Properties
+     * │           └── Parameters
+     * │               └── MyParameter
+     * │
+     * ├── MyNestedStack (Nested Stack Node - Path returned by this function)
+     * │   ├── ChildResource1
+     * │   │   └── Properties
+     * │   └── ChildResource2
+     * │       └── Properties
+     * ```
+     *
+     * @param nestedStackResourcePath - The path to the nested stack resource in the construct tree
+     * @param logicalId - The logicalId of the nested stack
+     * @returns The path to the nested stack wrapper node in the construct tree
+     */
+    public static getNestedStackPath(nestedStackResourcePath: string, logicalId: string): string {
```

Comment:
Nit: type aliaseses instead of `string` for these things can be helpful to clarify at type system level what's what.